protocols/300073/300073.ot2.apiv2.py

```diff
@@ -0,0 +1,385 @@
+        for that tip box. `pick_up()` will track tips for both pipettes if
+        applicable.
+
+        :param pipette: The pipette desired to pick up tip
+        as definited earlier in the protocol (e.g. p300, m20).
+        """
+        try:
+            pipette.pick_up_tip()
+        except protocol_api.labware.OutOfTipsError:
+            ctx.pause("Replace empty tip racks")
+            pipette.reset_tipracks()
+            pipette.pick_up_tip()
+
+    # helper functions
+    '''
+    Define any custom helper functions outside of the pipette scope here, using
+    the convention seen above.
+
+    e.g.
+
+    def remove_supernatant(vol, index):
+        """
+        function description
+
+        :param vol:
+
+        :param index:
+        """
+
+
+    '''
+    def transfer_tube_samples(volume, source_tubes, dest_wells):
+        nonlocal pipette, aspiration_height_mm, dispension_height_mm
+        nonlocal post_aspiration_wait
+
+        for s_well, d_well in zip(source_tubes, dest_wells):
+            pick_up(pipette)
+            pipette.aspirate(sample_aspiration_vol_ul,
+                             s_well.bottom(aspiration_height_mm))
+            ctx.delay(post_aspiration_wait)
+            pipette.dispense(sample_aspiration_vol_ul,
+                             d_well.bottom(dispension_height_mm))
+            pipette.drop_tip()
+
+    # reagents
+
+    '''
+    Define where all reagents are on the deck using the labware defined above.
+
+    e.g.
+
+    water = reservoir12.wells()[-1]
+    waste = reservoir.wells()[0]
+    samples = plate.rows()[0][0]
+    dnase = tuberack.wells_by_name()['A4']
+
+    '''
+
+    # plate, tube rack maps
+
+    '''
+    Define any plate or tube maps here.
+
+    e.g.
+
+    plate_wells_by_row = [well for row in plate.rows() for well in row]
+
+    '''
+    def add_quadrant_wells(list, columns, row_start_index, row_end_index):
+        """
+        Adds the wells in the given quadrant(rectangle) defined
+        by the columns and the start and end indices of the rows
+        col 1/row 1 ... col n/row 1
+        .  .            .
+        .       .       .
+        .            .  .
+        col 1/row n ... col n/row n
+        :param list: The list to append the wells in the quadrant to
+        :param colums: The columns that define the "x" interval of the quadrant
+        :param row_start_index: defines the start "y" interval
+        :param row_end_index: defines the end "y" interval
+        """
+        for col in columns:
+            for well in col[row_start_index:row_end_index]:
+                list.append(well)
+
+    # Map the target quadrants of the destination plate
+    targ_cols = plate.columns()
+    # Top left quadrant minus first 3 wells:
+    # The rectangle of A1-D1 to A6-D6 minus well A1 to A3
+    target_quadrant_1 = []
+    target_quadrant_1.append(targ_cols[0][3])  # Well D4
+    add_quadrant_wells(target_quadrant_1,
+                       targ_cols[1:6], 0, 4)
```

Comment:
I think this is wrong...
`targ_cols[0][3]` will equal `D1`
`...targ_cols[1:6], 0, 4` will append columns 2-6, rows A-D

So if my math is correct and I'm following, `target_quadrant_1` will contain `D1`, then `A2-D6`

Reply:
This is actually correct, A, B, C1 are left outside the sample well range for the tech to add controls to. So it should go D1, A2, B2, C2, D2, A3, B3 etc.

Reply:
okay sounds good... I thought A1, A2, and A3 were supposed to be excluded here, but it sounds like you have the correct layout then